### PR TITLE
[Event Hubs Client] Track Two: Second Preview (Spelling, Formatting, Minor Tweaks)

### PR DESCRIPTION
**Pull request by jsquire**
# Summary

The intent of these changes is to perform some minor clean up and formatting.

### Goals

- Spellcheck comments and strings for the project, fixing errors.

- Perform an automated formatting run over the solution.

- Convert `new byte[0]` calls to `Array.Empty<byte>()` in event translation areas to reduce unnecessary allocations.

# Last Upstream Rebase

Thursday, July 25, 2019  8:43am (EDT)

# Resources

- [Azure Messaging Exception Documentation](https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-messaging-exceptions)
- [.NET Event Hubs Client: Second Preview Design Proposal](https://github.com/jsquire/azure-sdk-for-net/blob/master/sdk/eventhub/Azure.Messaging.EventHubs/design/proposal-event-hubs-second-preview.md)

# Related and Follow-Up Issues

- [[Epic] Release Event Hubs Track 2 Library Preview 2](https://github.com/Azure/azure-sdk-for-net/issues/6752) (#6752)

### CODE REVIEW

File: sdk/eventhub/Azure.Messaging.EventHubs/tests/Compatibility/TrackOneRetryPolicyTests.cs [hunk 1]
```diff
@@ -60,7 +60,7 @@ public void RetryIsInvokedForRetriableNonEventHubsException()
                 .Setup(policy => policy.CalculateRetryDelay(It.Is<Exception>(value => Object.ReferenceEquals(value, lastException)), It.Is<int>(value => value == retryCount)))
                 .Returns(expectedInterval);
 
-            Assert.That(TrackOne.RetryPolicy.IsRetryableException(lastException), Is.True, "The operation cancelled exception should be considered as retriable by the TrackOne.RetryPolicy.");
+            Assert.That(TrackOne.RetryPolicy.IsRetryableException(lastException), Is.True, "The operation canceled exception should be considered as retriable by the TrackOne.RetryPolicy.");
```
Comment (kinelski) [inline sdk/eventhub/Azure.Messaging.EventHubs/tests/Compatibility/TrackOneRetryPolicyTests.cs]: Interestingly enough, it seems both spellings are correct.

File: sdk/eventhub/Azure.Messaging.EventHubs/tests/EventHubClient/EventHubClientTests.cs [hunk 2]
```diff
@@ -275,7 +275,7 @@ public void ConstructorWithConnectionStringValidatesOptions()
         public void ConstructorWithExpandedArgumentsValidatesOptions()
         {
             var invalidOptions = new EventHubClientOptions { TransportType = TransportType.AmqpTcp, Proxy = Mock.Of<IWebProxy>() };
-            Assert.That(() => new EventHubClient("host", "path", Mock.Of<TokenCredential>(), invalidOptions), Throws.ArgumentException, "The cexpanded argument onstructor should validate client options");
+            Assert.That(() => new EventHubClient("host", "path", Mock.Of<TokenCredential>(), invalidOptions), Throws.ArgumentException, "The expanded argument constructor should validate client options");
```
Comment (kinelski) [inline sdk/eventhub/Azure.Messaging.EventHubs/tests/EventHubClient/EventHubClientTests.cs]: At least we got all the letters right at first.

**Review by kinelski — APPROVED** [review]
LGTM.

File: sdk/eventhub/Azure.Messaging.EventHubs/tests/Compatibility/TrackOneRetryPolicyTests.cs (hunk 1 above)
Comment (jsquire) [reply sdk/eventhub/Azure.Messaging.EventHubs/tests/Compatibility/TrackOneRetryPolicyTests.cs]: I've always used two "l"s, but given how poorly my typos and spellings have been, I was trusting the dictionary.   :)

File: sdk/eventhub/Azure.Messaging.EventHubs/tests/Amqp/TypeExtensionsTests.cs [hunk 3]
```diff
@@ -76,7 +76,7 @@ public void ToAmqpPropertyMapsKnownTypes(Type knownType)
         }
 
         /// <summary>
-        ///   Veriies functionality of the <see cref="TypeExtensions.ToAmqpPropertyType" />
```
Comment (conniey) [inline sdk/eventhub/Azure.Messaging.EventHubs/tests/Amqp/TypeExtensionsTests.cs]: I am _pretty sure_ this is the correct spelling. 😄

**Pull request merged**